### PR TITLE
refactor(backend): change image field of book

### PR DESCRIPTION
**Pull request by GotaSeptimiuAndrei**
- related issue #18
- changed the image field to store url
- created a bucket in aws to store images
- refactored book controller and service accordingly

### CODE REVIEW

**Comment by GotaSeptimiuAndrei** [comment]
Screenshot 2024-11-05 211628

File: backend/src/main/java/com/backend/Controller/BookController.java [hunk 1]
```diff
@@ -40,81 +39,115 @@ public ResponseEntity<Book> getBookById(@PathVariable Long id) {
...
-	public ResponseEntity<Book> createBook(@RequestBody Book book) {
-		if (book.getImage() != null) {
-			try {
-				book.setImage(Base64.getDecoder().decode(new String(book.getImage())));
-			}
-			catch (IllegalArgumentException e) {
-				logger.error("Invalid Base64 input for image", e);
-				return ResponseEntity.badRequest().body(null);
-			}
+	public ResponseEntity<String> createBook(@RequestParam("title") String title,
```
Comment (DeceaCezara) [inline backend/src/main/java/com/backend/Controller/BookController.java:42]: De ce ai inlocuit body cu query parameters? Ar trebui sa trimitem datele folosind body

File: backend/src/main/java/com/backend/Controller/BookController.java [hunk 2]
```diff
@@ -40,81 +39,115 @@ public ResponseEntity<Book> getBookById(@PathVariable Long id) {
...
-			updatedBook.setPrice(bookDetails.getPrice());
-			updatedBook.setImage(bookDetails.getImage());
-			updatedBook.setRating(bookDetails.getRating());
-			updatedBook.setAvailable_count(bookDetails.getAvailable_count());
-			updatedBook.setCategory(bookDetails.getCategory());
-			return ResponseEntity.ok(bookService.saveBook(updatedBook));
-		}
-		else {
-			return ResponseEntity.notFound().build();
+	public ResponseEntity<String> updateBook(
```
Comment (DeceaCezara) [inline backend/src/main/java/com/backend/Controller/BookController.java:69]: Acelasi lucru aici

File: backend/src/main/java/com/backend/Controller/BookController.java [hunk 3]
```diff
@@ -40,81 +39,115 @@ public ResponseEntity<Book> getBookById(@PathVariable Long id) {
...
-			}
-			if (bookDetails.getRating() != null) {
-				existingBook.setRating(bookDetails.getRating());
-			}
-			if (bookDetails.getAvailable_count() != null) {
-				existingBook.setAvailable_count(bookDetails.getAvailable_count());
-			}
-			if (bookDetails.getCategory() != null) {
-				existingBook.setCategory(bookDetails.getCategory());
+	public ResponseEntity<String> partiallyUpdateBook(
```
Comment (DeceaCezara) [inline backend/src/main/java/com/backend/Controller/BookController.java:119]: Aceeasi intrebare

File: backend/src/main/java/com/backend/Controller/BookController.java (hunk 1 above)
Comment (GotaSeptimiuAndrei) [reply backend/src/main/java/com/backend/Controller/BookController.java]: Am inlocuit cu model attribute, nu putem folosi RequestBody pt ca nu putem trimite fisierul in json

**Pull request merged**